modelci/hub/converter/to_tvm.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+#  Copyright (c) NTU_CAP 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from pathlib import Path
+from typing import Iterable, List, Optional, Callable
+
+import tvm
+import numpy 
+import torch
+import torch.jit
+from torchvision import transforms
+from modelci.utils import Logger
+
+from tvm import relay
+from tvm.contrib.download import download_testdata
+# from modelci.types.bo import IOShape
+
+from pathlib import Path
+from PIL import Image
+
+logger = Logger('converter', welcome=False)
+
+
+class TVMConverter(object):
+    """Convert model to TVM format."""
+
+    supported_framework = ["pytorch"]
+
+    @staticmethod
+    def from_pytorch(model: torch.nn.Module, save_path: Path, img_path, input_shape, input_name, processing, opt_level, override: bool = False):
+        """Convert a PyTorch nn.Module into TorchScript.
+        """
+        if save_path.with_suffix('.zip').exists():
+            if not override:  # file exists yet override flag is not set
+                print('Use cached model')
+                return True
+        model.eval()
+        try:
+            input_data = torch.randn(input_shape)
+            traced_model = torch.jit.trace(model, input_data).eval()
+            
+            # image for testing 
+            img = Image.open(img_path).resize((224, 224))
```

Comment:
We do not only support image models.

Please check the other converters and make this part more general

Reply:
The general idea is to generate fake input from the `IOShape` provided to you.